### PR TITLE
chore(#1606): always use file system watchers

### LINKED ISSUE #1606: Finalise filesystem_watchers
Remove any codepaths that are conditional on either of these options.

Silently remove these options.

Update: there are genuine use cases for disabled watchers. Resolve remaining issues such as large directories.

Tasks:
- [x] add `filesystem_watchers.ignore`
- [x] tear down watcher on failure, notifying the user once

### PR DESCRIPTION
**Pull request by alex-courtis**
resolves #1606 

Watchers are stable and always used.

Any remaining reloading is user invoked or necessary for actions such as filtering.

I would greatly appreciate some testing on this one. I'll soak for a week.

### CODE REVIEW

File: lua/nvim-tree/actions/init.lua [hunk 1]
```diff
@@ -393,7 +393,6 @@ function M.setup(opts)
   require("nvim-tree.actions.node.open-file").setup(opts)
   require("nvim-tree.actions.root.change-dir").setup(opts)
   require("nvim-tree.actions.fs.create-file").setup(opts)
-  require("nvim-tree.actions.fs.rename-file").setup(opts)
```
Comment (gegoune) [inline lua/nvim-tree/actions/init.lua:396]: Was that removed accidentally?

File: lua/nvim-tree/marks/init.lua [hunk 2]
```diff
@@ -70,7 +70,6 @@ end
 
 function M.setup(opts)
   vim.fn.sign_define(SIGN_NAME, { text = opts.renderer.icons.glyphs.bookmark, texthl = "NvimTreeBookmark" })
-  require("nvim-tree.marks.bulk-move").setup(opts)
```
Comment (gegoune) [inline lua/nvim-tree/marks/init.lua:73]: Was that removed accidentally?

**Review by gegoune — COMMENTED** [review]
LGTM except for two questions. Quick testing revealed not problems.

File: lua/nvim-tree/marks/init.lua (hunk 2 above)
Comment (alex-courtis) [reply lua/nvim-tree/marks/init.lua:73]: That setup function was removed as it was no longer needed.

File: lua/nvim-tree/actions/init.lua (hunk 1 above)
Comment (alex-courtis) [reply lua/nvim-tree/actions/init.lua:396]: That setup function was removed as it was no longer needed.

File: lua/nvim-tree/legacy.lua [hunk 3]
```diff
@@ -313,6 +321,9 @@ function M.migrate_legacy_options(opts)
   -- silently move
   refactored(opts)
 
+  -- silently clean unnecessary options that are now default behaviour
+  unnecessary(opts)
```
Comment (alex-courtis) [inline lua/nvim-tree/legacy.lua:325]: Contemplating a mechanism of informing users, however users do get upset when messages are displayed on startup.

Perhaps some sort of one-shot mechanism or a way to disable messages.

Comment (gegoune) [reply lua/nvim-tree/legacy.lua:325]: Perhaps something like `vim.notify_once()` could be used? I really dont mind those messages either, would rather know about changes I need to make than have broken setup.

Comment (alex-courtis) [reply lua/nvim-tree/legacy.lua:325]: Today I Learned `notify_once`.

I was hopingi it persists across sessions, however sadly not.

#1667

**Comment by alex-courtis** [comment]
Watchers disabled is desirable.

**Pull request closed without merging**